src/animation/Animation.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Animation } from './Animation.js';
+
+
+class TestAnimation extends Animation {
+  public setUpCalled = false;
+  public tearDownCalled = false;
+  public animationCalls: Array<[number, number]> = [];
+  public beforeCalled = false;
+  public afterCalled = false;
+  
+  constructor() {
+    super(
+      (elapsedTime: number, deltaTime: number) => {
+        this.animationCalls.push([elapsedTime, deltaTime]);
+      },
+      { 
+        before: (() => { this.beforeCalled = true; }) as any,
+        after: (() => { this.afterCalled = true; }) as any
+      }
+    );
+  }
+
+  setUp(): void {
+    super.setUp();
+    this.setUpCalled = true;
+  }
+
+  tearDown(): void {
+    super.tearDown();
+    this.tearDownCalled = true;
+  }
+}
+
+// Test suite
+describe('Animation', () => {
```

Comment:
I think I mentioned this before, but these functions should be imported from vitest